tests/phpunit/tests/image/functions.php

```diff
@@ -248,6 +248,220 @@ public function test_is_gd_image_invalid_types() {
+
+		$this->assertNotEmpty( $backup_sizes );
+		$this->assertIsArray( $backup_sizes );
+
+		$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+		$this->assertIsArray( $backup_sources );
+		$this->assertArrayHasKey( 'full-orig', $backup_sources );
+		$this->assertSame( $metadata['sources'], $backup_sources['full-orig'] );
+
+		foreach ( $backup_sizes as $size => $properties ) {
+			$size_name = str_replace( '-orig', '', $size );
+
+			if ( 'full-orig' === $size ) {
+				continue;
+			}
+
+			$this->assertArrayHasKey( 'sources', $properties );
+			$this->assertSame( $metadata['sizes'][ $size_name ]['sources'], $properties['sources'] );
+		}
+	}
+
+	/**
+	 * Restore the sources array from the backup when an image is edited
+	 *
+	 * @ticket 55443
+	 */
+	public function test_restore_the_sources_array_from_the_backup_when_an_image_is_edited() {
+		require_once ABSPATH . 'wp-admin/includes/image-edit.php';
+
+		$attachment_id = self::factory()->attachment->create_upload_object( DIR_TESTDATA . '/images/canola.jpg' );
+
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+
+		$_REQUEST['action']  = 'image-editor';
+		$_REQUEST['context'] = 'edit-attachment';
+		$_REQUEST['postid']  = $attachment_id;
+		$_REQUEST['target']  = 'all';
+		$_REQUEST['do']      = 'save';
+		$_REQUEST['history'] = '[{"r":-90}]';
+
+		wp_save_image( $attachment_id );
+
+		$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+
+		$this->assertArrayHasKey( 'full-orig', $backup_sources );
+		$this->assertIsArray( $backup_sources['full-orig'] );
+		$this->assertSame( $metadata['sources'], $backup_sources['full-orig'] );
+
+		wp_restore_image( $attachment_id );
+
+		$metadata               = wp_get_attachment_metadata( $attachment_id );
+		$updated_backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
+
+		$this->assertSame( $backup_sources['full-orig'], $metadata['sources'] );
+		$this->assertNotSame( $backup_sources, $updated_backup_sources );
+		$this->assertCount( 1, $backup_sources );
+		$this->assertCount( 2, $updated_backup_sources );
+
+		$backup_sizes = get_post_meta( $attachment_id, '_wp_attachment_backup_sizes', true );
+		foreach ( $backup_sizes as $size_name => $properties ) {
+			// We are only interested in the original filenames to be compared against the backup and restored values.
+			if ( false === strpos( $size_name, '-orig' ) ) {
+				continue;
+			}
+
+			$size_name = str_replace( '-orig', '', $size_name );
+			// Full name is verified above.
+			if ( 'full' === $size_name ) {
+				continue;
+			}
+
+			$this->assertArrayHasKey( $size_name, $metadata['sizes'] );
+			$this->assertArrayHasKey( 'sources', $metadata['sizes'][ $size_name ] );
+			$this->assertSame( $properties['sources'], $metadata['sizes'][ $size_name ]['sources'] );
+		}
+	}
+
+	/**
+	 * Prevent to backup the full size image if only the thumbnail is edited
+	 *
+	 * @test
```

Comment:
Remove this annotation as a function start with `test_`. Add ticket number `@ticket 55443`